doc/examples/applications/plot_3d_image_processing.py

```diff
@@ -94,7 +94,7 @@
 
 def show_plane(ax, plane, cmap="gray", title=None):
     ax.imshow(plane, cmap=cmap)
-    ax.axis("off")
+    ax.set_axis_off()
```

Comment:
I'm curious why you changed some of these but not all? E.g. in [`plot_general.py`](https://github.com/scikit-image/scikit-image/blob/9107d523a899b8d3767630941ac28f438bf875d2/doc/examples/data/plot_general.py#L58)? 
To be clear I don't mind either way, both methods are part of the OO API. But it does blow up the diff to quite a lot. Keeping diffs small can help speeding things along as getting big PRs merged can sometimes be a slog.

The same thing applies with 
- `fig, (ax1, ax2) = plt.subplots(...)` and 
- `fig, axes = plt.subplots(...)` 

which were changed to `fig, ax = plt.subplots(...)`.

Reply:
I have not made yet to the file plot_general.py. I only changed the files in the folder applications/ so far.
I left (ax1, ax2) and similar in a few instances where I feared to break the docs making a change. E.g. in plot_image_comparison.py, GridSpec is used to arrange the plots in a more complex layout and changing fig, (ax1, ax2, ax3) into fig, ax generated an error I could not solve.

Reply:
I don't see any reason, in general, for changing `fig, axes = plt.subplots(...)` or `fig, (ax1, ax2) = plt.subplots(...)` into `fig, ax = plt.subplots(...)`. Using `ax1` and `ax2` may even be more convenient and readable than `ax[0]` and `ax[1]`. Anyway, these changes have nothing to do with the OO style vs pyplot-style issue.

Reply:
> I only changed the files in the folder applications/ so far.

Ah I should have noticed. Then let's keep this PR focused on only that directory. And we can do follow-ups for the other directories.

I concur with @mkcor but how do we go forward? I'd say to minimize work on all sides let's keep the instances where you changed `.axis("off")` to `.set_axis_off()` but let's not change more. They are easy to review but the consistency is not really a big improvement in this case.

For the `(ax1, ax2)` vs `ax` issue, I also don't see a big difference. Let's keep the ones you changed and not change more going forward. @mkcor agreed?

Reply:
Sorry, now I am bit lost about the changes that I still need to do for this pull request and the general guidelines you wish to see implemented. I try to recap:

**For this pull request**

- Change plt.close() in fig.close()
- Change plt.tight_layout() in fig.tight_layout()
- I can revert the set_axis_off()  in axis("off") when this was the original use (just say what you prefer)
- I think I already reverted all fig, ax = plt.subplots() in _, ax = plt.subplots(), but if I will check if I missed any.

**In general**

-  _, ax = plt.subplots() does not need to be changed in fig, ax = plt.subplots()
- fig, axes = plt.subplots() does not need to be changed in fig, ax = plt.subplots()
- fig, (ax1, ax2) = plt.subplots() and similar do not need to be changed in fig, ax = plt.subplots()
- Axis notations ax1, etc. and ax[1], etc. are both acceptable
- set_axis_off()  and axis("off") are both acceptable
- plt.close() should be changed in fig.close()
- plt.tight_layout() should be changed in fig.tight_layout()
- plt.plot() must be replaced by plt.show()
- Editorial changes in docs (capital letters in titles, etc.) are better left for future interventions.

Please provide your feedback.

Reply:
Hi both,

@GParolini said (private communication) that she would away in the next days and we could take over the PR. For future reference:

> let's keep the instances where you changed .axis("off") to .set_axis_off() but let's not change more

Agreed.

> For the `(ax1, ax2)` vs `ax` issue, I also don't see a big difference. Let's keep the ones you changed and not change more going forward. @mkcor agreed?

Hmm, I would revert these, since a) it's unrelated to the OO vs pyplot-style and b) `(ax1, ax2)` is correct. I can take care of it (see above).

> Change plt.close() to fig.close()

Yes, for sure, since plt.close() is pyplot-style, so it corresponds to the issue we're addressing with this PR.

> Change plt.tight_layout() in fig.tight_layout()

idem

I don't think I need to go through all the bullet points, since the issue is unambiguous: Everything pyplot-style needs to be changed into OO style, period.

Reply:
> plt.close() should be changed in fig.close()

Apparently this doesn't work (see [failure](https://github.com/scikit-image/scikit-image/actions/runs/8360293702/job/22885635348?pr=7346)); reverted 09df1bc48cfc75da.